packages/edit-site/src/store/selectors.js

```diff
@@ -288,6 +288,31 @@ export function getEditorMode( state ) {
 	return __unstableGetPreference( state, 'editorMode' );
 }
 
+/**
+ * Returns the post currently being edited in its last known saved state, not
+ * including unsaved edits. Returns an object containing relevant default post
+ * values if the post has not yet been saved.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {Object} Post object.
+ */
+export const getCurrentTemplate = createRegistrySelector(
+	( select ) => ( state ) => {
+		const templateType = getEditedPostType( state );
+		const templateId = getEditedPostId( state );
+		const template = templateId
+			? select( coreDataStore ).getEntityRecord(
+					'postType',
+					templateType,
+					templateId
+			  )
+			: null;
+
+		return template;
+	}
+);
```

Comment:
@draganescu, maybe we should remove this new selector and move logic inside the new hook?

The hooks are private most of the time, and we can easily remove them.

Reply:
I think a seletor named `getCurrentTemplate` is good! I am less inclined to give this one up :)

Reply:
Can we use this selector in other places? There's also the `useEditedEntityRecord` hook that returns similar data.

See: https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/use-edited-entity-record/index.js

Reply:
Hmm, I didn't know about `useEditedEntityRecord` - I am not a fan of building hooks to avoid obvious selectors, but OTOH let's use what exists for now. Updated in [d6712fd](https://github.com/WordPress/gutenberg/pull/45215/commits/d6712fd8efa3f6e076c5f1f554a2af0af86f7ac9)

Reply:
The hooks that aren't exported from the packages are easy to modify or remove. But, unfortunately, every selector is part of the public API[^1], and we've to maintain it "forever."

[^1]: Unless marked private via the new package.

Reply:
I know about that, and I agree to avoid cluttering the public API with what might be temporary :) needs but usually getters setters for things that will likely never change (templates and template parts for instance) should be added on 1st occasion. What I am saying is `getCurrentTemplate` should have been added when `useEditedEntityRecord` was needed because knowing the currently edited template is a default need in a "template editor" 😁